common/src/types.rs

```diff
@@ -119,14 +123,16 @@ pub struct TxDelegate {
     pub delegatee: PublicKey,
     /// Whether to delegate the governance voting power too.
     pub governance: bool,
-    pub proof: TypedSignature<(PublicKey, PublicKey, bool, BlockHeight)>,
+    pub proof: DelegationSignature,
     pub timestamp: Timestamp,
 }
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub struct TxUndelegate {
     pub delegator: PublicKey,
-    pub proof: TypedSignature<(PublicKey, BlockHeight)>,
+    pub delegatee: PublicKey,
+    pub governance: bool,
```

Comment:
Why did you add these fields?

Reply:
As we discussed before, the delegation has two branches where the `governance` is true that results in delegating both `governance` and `consensus` while only `consensus` is delegated when `governance` option is false. When considering this requirement, should the same logic be applied to TxUndelegate as well? What do you think?

Reply:
`governance` field might be necessary in that case but I don't understand why delegatee field is neither still.

Btw, check this [conversation](https://discord.com/channels/941206496203669545/990866117960531969/1074616134164287518) where the specification is discussed to undelegate as a whole.

Please revert the above changes.

Reply:
checked the discord thread, and removed the field.